src/GovUk.Education.ExploreEducationStatistics.Content.Services/ReleaseService.cs

```diff
@@ -24,37 +25,83 @@ public class ReleaseService : IReleaseService
         private readonly IFileStorageService _fileStorageService;
         private readonly IMethodologyService _methodologyService;
         private readonly IUserService _userService;
+        private readonly IPublicationService _publicationService;
+        private readonly IMapper _mapper;
 
         public ReleaseService(
             IPersistenceHelper<ContentDbContext> persistenceHelper,
             IFileStorageService fileStorageService,
             IMethodologyService methodologyService,
-            IUserService userService)
+            IUserService userService,
+            IPublicationService publicationService,
+            IMapper mapper)
+
         {
             _persistenceHelper = persistenceHelper;
             _fileStorageService = fileStorageService;
             _methodologyService = methodologyService;
             _userService = userService;
+            _publicationService = publicationService;
+            _mapper = mapper;
         }
 
-        public async Task<Either<ActionResult, ReleaseViewModel>> Get(string publicationPath, string releasePath)
+        public async Task<Either<ActionResult, ReleaseViewModel>> Get(string publicationSlug, string? releaseSlug = null)
         {
-            return await CreateFromCachedPublicationAndRelease(publicationPath, releasePath,
-                    (publication, release) => new ReleaseViewModel(release, publication))
-                .OnSuccessCombineWith(model => _methodologyService.GetSummariesByPublication(model.Publication.Id))
-                .OnSuccess(viewModelAndMethodologies =>
+            return await _persistenceHelper
+                .CheckEntityExists<Publication>(query => query
+                    .Where(p => p.Slug == publicationSlug))
+                .OnSuccess(publication => _methodologyService.GetSummariesByPublication(publication.Id))
+                .OnSuccess(async methodologies =>
                 {
-                    var (viewModel, methodologies) = viewModelAndMethodologies;
-                    viewModel.Publication.Methodologies = methodologies;
-                    return viewModel;
+                    var publicationTask = _publicationService.Get(publicationSlug);
+                    var releaseTask = FetchCachedRelease(publicationSlug, releaseSlug);
+
+                    await Task.WhenAll(publicationTask, releaseTask);
+
+                    if (releaseTask.Result.IsRight
```

Comment:
I think we could improve on this by changing back to `OnSuccessCombineWith`. Elsewhere I've said that I think we could be getting the Publication by Id rather than slug so that kind of invalidates what I'm about to say as we'd need to look up the Publication by slug still somewhere, but this method could have started with `_publicationService.Get` and then `.OnSuccessCombineWith(model => _methodologyService.GetSummariesByPublication(model.Publication.Id))` rather than start with `persistenceHelper.CheckEntityExists<Publication>(query => query.Where(p => p.Slug == publicationSlug))`.

That's just duplicating what `_publicationService.Get` is already doing.